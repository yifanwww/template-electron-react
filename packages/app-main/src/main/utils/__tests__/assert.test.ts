import { describe, expect, it } from '@jest/globals';
import { AssertionError } from 'node:assert';

import { assertIsNever } from '../assert';

describe(`Test fn \`${assertIsNever.name}\``, () => {
    it('should throw an error', () => {
        expect(() => assertIsNever('hello world' as never)).toThrow(AssertionError);
        expect(() => assertIsNever('hello world' as never)).toThrowErrorMatchingSnapshot();
    });
});
