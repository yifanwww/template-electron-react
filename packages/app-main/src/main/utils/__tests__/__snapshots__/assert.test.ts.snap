// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test fn \`assertIsNever\` should throw an error 1`] = `""hello world" should be "never" type."`;
